common/cpp/crypto/sig_private_key.cpp

```diff
@@ -204,7 +188,7 @@ void pcrypto::sig::PrivateKey::Generate() {
         throw Error::RuntimeError(msg);
     }
 
-    EC_GROUP_ptr ec_group(EC_GROUP_new_by_curve_name(constants::CURVE),
+    EC_GROUP_ptr ec_group(EC_GROUP_new_by_curve_name(NID_secp256k1),
```

Comment:
We may want to use different curves so it is better to have a corresponding definition in an crypto lib generic header file. See my generic PR comment #3.

Reply:
@EugeneYYY: I restoerd constants::CURVE and moved it to an internal header file, crypto_shared.h so it will not be exposed to non-crypto Avalon code.

Reply:
@EugeneYYY: I added CRYPTOLIB_OPENSSL and CRYPTOLIB_MBEDTLS to private header file `crypto_shared.h` and `tests/Makefile`.

I did not add `CRYPTOLIB*` to crypto.h because it is used by crypto code in other parts of Avalon and the API should be library-independent.

I updated the PR, so please re-review.